packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -128,6 +129,31 @@ abstract class OneSequenceGestureRecognizer extends GestureRecognizer {
     super.dispose();
   }
 
+  /// The team that this recognizer belongs to, if any.
+  ///
+  /// If [team] is null, this recognizer competes directly in the
+  /// [GestureArenaManager] to recognize a sequence of pointer events as a
+  /// gesture. If [team] is non-null, this recognizer competes in the arena in
+  /// a group with other recognizers on the same team.
+  ///
+  /// A recognizer can be assigned to a team only when it is not participating
+  /// in the arena. For example, a common time to assign a recognizer to a team
+  /// is shortly after creating the recognizer.
+  GestureArenaTeam get team => _team;
+  GestureArenaTeam _team;
+  set team(GestureArenaTeam value) {
+    assert(_entries.isEmpty);
+    assert(_trackedPointers.isEmpty);
+    assert(_team == null);
+    _team = value;
+  }
+
+  GestureArenaEntry _addPointerToArena(int pointer) {
+    if (_team != null)
+      return _team.add(pointer, this);
+    return GestureBinding.instance.gestureArena.add(pointer, this);
+  }
+
```

Comment:
Another option here that might be cleaner would be to have a single interface implemented by both GestureArena and GestureArenaTeam, and have all gestures always be assigned an "owner", which is used here.

That would avoid the GestureRecognizer needing to know about GestureArenaTeam. It seems weird for it to know about teams. It would also remove another use of a global static, and would therefore allow you to run multiple arenas at the same time (which might be useful when you have multiple pipelines or something).